collectors/wt_client_collector.go

```diff
@@ -0,0 +1,86 @@
+package collectors
+
+import (
+	"context"
+	"fmt"
+	"encoding/hex"
+
+	"github.com/lightninglabs/lndclient"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// WtClientCollector is a collector that will export watchtower client related
+// metrics
+type WtClientCollector struct {
+	lnd *lndclient.LndServices
+
+	numBackupsDesc *prometheus.Desc
+	numPendingBackupsDesc *prometheus.Desc
+
+	// errChan is a channel that we send any errors that we encounter into.
+	// This channel should be buffered so that it does not block sends.
+	errChan chan<- error
+}
+
+// NewWalletCollector returns a new instance of the WalletCollector.
+func NewWtClientCollector(lnd *lndclient.LndServices,
+	errChan chan<- error) *WtClientCollector {
+
+	return &WtClientCollector{
+		lnd: lnd,
+		numBackupsDesc: prometheus.NewDesc(
+			"lnd_wt_client_num_backups", "watchtower client number of backups",
+			[]string{
+				"tower_pubkey",
+			}, nil,
+		),
+		numPendingBackupsDesc: prometheus.NewDesc(
+			"lnd_wt_client_num_pending_backups", "watchtower client number of pending backups",
+			[]string{
+				"tower_pubkey",
+			}, nil,
+		),
+
+		errChan: errChan,
+	}
+}
+
+// Describe sends the super-set of all possible descriptors of metrics
+// collected by this Collector to the provided channel and returns once the
+// last descriptor has been sent.
+//
+// NOTE: Part of the prometheus.Collector interface.
+func (c *WtClientCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- c.numBackupsDesc
+	ch <- c.numPendingBackupsDesc
+}
+
+// Collect is called by the Prometheus registry when collecting metrics.
+//
+// NOTE: Part of the prometheus.Collector interface.
+func (c *WtClientCollector) Collect(ch chan<- prometheus.Metric) {
+	towers, err := c.lnd.WtClient.ListTowers(context.Background(), true, false)
+	if err != nil {
+		c.errChan <- fmt.Errorf("WtClientCollector ListTowers failed "+
+			"with: %v", err)
+		return
+	}
+	
+	for _, tower := range towers {
+		pubkey := hex.EncodeToString(tower.Pubkey)
+		numBackups := uint32(0)
+		numPendingBackups := uint32(0)
+		for _, sessionInfo := range tower.SessionInfo {
+			for _, session := range sessionInfo.Sessions {
+				numBackups += session.NumBackups
+				numPendingBackups += session.NumPendingBackups
+			}
+		}
+		ch <- prometheus.MustNewConstMetric(
+			c.numBackupsDesc, prometheus.GaugeValue, float64(numBackups), pubkey,
```

Comment:
This metric could decrease as well if the tower gets deleted, so it looks good to me using a gauge here. Great you labeled this by pubkey.